test-data/unit/check-python310.test

```diff
@@ -1528,3 +1528,20 @@ class A:
 class B:
     def __enter__(self) -> B: ...
     def __exit__(self, x, y, z) -> None: ...
+
+[case testOverrideMatchArgs]
+class AST:
+   __match_args__ = ()
+
+class stmt(AST): ...
+
+class AnnAssign(stmt):
+   __match_args__ = ('target', 'annotation', 'value', 'simple')
+
+reveal_type(AST.__match_args__)  # N: Revealed type is "Tuple[]"
+reveal_type(stmt.__match_args__)  # N: Revealed type is "Tuple[]"
+reveal_type(AnnAssign.__match_args__)  # N: Revealed type is "Tuple[builtins.str, builtins.str, builtins.str, builtins.str]"
+
```

Comment:
Maybe add a `match` statement to confirm that AnnAssign's `__match_args__` works correctly?

Reply:
Added.